tests/baselines/reference/emitBOM.js

```diff
@@ -2,14 +2,20 @@
 // JS and d.ts output should have a BOM but not the sourcemap
 var x;
 
+tests/cases/compiler/emitBOM.js(1,1): error TS1434: Unexpected keyword or identifier.
 tests/cases/compiler/emitBOM.js(1,2): error TS1127: Invalid character.
+tests/cases/compiler/emitBOM.js(1,2): error TS1128: Declaration or statement expected.
 tests/cases/compiler/emitBOM.js(1,3): error TS1127: Invalid character.
 
 
-==== tests/cases/compiler/emitBOM.js (2 errors) ====
+==== tests/cases/compiler/emitBOM.js (4 errors) ====
     ï»¿// JS and d.ts output should have a BOM but not the sourcemap
+    ~
+!!! error TS1434: Unexpected keyword or identifier.
      
 !!! error TS1127: Invalid character.
+     ~
+!!! error TS1128: Declaration or statement expected.
       
 !!! error TS1127: Invalid character.

```

Comment:
This seems like a slight regression. Is there any way to clean this up?

Reply:
Hmm, this is tricky... if we forget about the existence of BOMs then the parser is working as it should: `ï` is a valid character but `»` and `¿` are not. I guess this will need to hardcode in that it shouldn't complain if this is specifically the BOM at the top of the file?

I don't see a way to avoid the introduced parsing logic if the scanner has identified an invalid character 😕

Reply:
As part of a052314 the new `parseErrorForMissingSemicolonAfter` function won't give an error if the token is `SyntaxKind.Unknown`.